ReactAndroid/src/main/java/com/facebook/react/modules/blob/BlobModule.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.modules.blob;
+
+import android.content.ContentResolver;
+import android.content.res.Resources;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.support.annotation.Nullable;
+import android.webkit.MimeTypeMap;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.common.MapBuilder;
+import com.facebook.react.module.annotations.ReactModule;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@ReactModule(name = "BlobModule")
+public class BlobModule extends ReactContextBaseJavaModule {
+
+
+  public BlobModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+  }
+
+  @Override
+  public String getName() {
+    return "BlobModule";
+  }
+
+  @Override
+  @Nullable
+  public Map getConstants() {
+    // The application can register BlobProvider as a ContentProvider so that blobs are resolvable.
+    // If it does, it needs to tell us what authority was used via this string resource.
+    Resources resources = getReactApplicationContext().getResources();
+    String packageName = getReactApplicationContext().getPackageName();
+    int resourceId = resources.getIdentifier("blob_provider_authority", "string", packageName);
+    if (resourceId == 0) {
+      return null;
+    }
+
+    return MapBuilder.of(
+      "BLOB_URI_SCHEME", "content",
+      "BLOB_URI_HOST", resources.getString(resourceId));
+  }
+
+  private static Map<String, byte[]> sBlobs = new HashMap<>();
```

Comment:
The store method needs to access this, since store method is static, this has to be static. Other parts of native code access the store method without having to initialise a new instance as this is shared globally.